File: Daily Challenge/April 2025/Minimum Operations to Make Array Values Equal to K.cpp
class Solution {
public:
  int minOperations(vector<int>& nums, int k) {
    set<int> s;
    for (int x : nums) {
      s.insert(x);
    }

    int operations = 0;
    for (int x : s) {
      if (x < k) return -1;
      if (x > k) operations++;
    }

    return operations;
  }
};

File: Daily Challenge/April 2025/Minimum Operations to Make Array Values Equal to K.cs
public class Solution {
  public int MinOperations(int[] nums, int k) {
    var set = new HashSet<int>();
    foreach (var num in nums) {
      set.Add(num);
    }

    int operations = 0;
    foreach (var num in set) {
      if (num < k) return -1;
      if (num > k) operations++;
    }

    return operations;
  }
}